src/aotcompile.cpp

```diff
@@ -2386,23 +2386,22 @@ void jl_dump_native_impl(void *native_code,
     }
 }
 
-
 // sometimes in GDB you want to find out what code would be created from a mi
-extern "C" JL_DLLEXPORT_CODEGEN jl_code_info_t *jl_gdbdumpcode(jl_method_instance_t *mi)
+extern "C" JL_DLLEXPORT_CODEGEN jl_code_instance_t *jl_gdbdumpcode(jl_method_instance_t *mi)
 {
     jl_llvmf_dump_t llvmf_dump;
     size_t world = jl_current_task->world_age;
     JL_STREAM *stream = (JL_STREAM*)STDERR_FILENO;
 
-    jl_code_info_t *src = jl_gdbcodetyped1(mi, world);
-    JL_GC_PUSH1(&src);
+    jl_code_instance_t *ci = jl_type_infer(mi, world, SOURCE_MODE_GET_SOURCE);
+    JL_GC_PUSH1(&ci);
 
     jl_printf(stream, "---- dumping IR for ----\n");
     jl_static_show(stream, (jl_value_t*)mi);
     jl_printf(stream, "\n----\n");
 
     jl_printf(stream, "\n---- unoptimized IR ----\n");
-    jl_get_llvmf_defn(&llvmf_dump, mi, src, 0, false, jl_default_cgparams);
+    jl_get_llvmf_defn(&llvmf_dump, ci, NULL, 0, false, jl_default_cgparams);
```

Comment:
It is not legal now to pass NULL here as the second argument, as codegen (intentionally) lacks the ability now to do inference itself